### DISCUSSION TITLE
How do i assign the value of a command in the config ?

### DISCUSSION THREAD

**Discussion by xd003** [polybar/polybar#2776] [General]
I am using the [xwindow](https://github.com/polybar/polybar/wiki/Module:-xwindow) module . I want it to only display the Focused App's Name 
( Eg Displaying Firefox & not the title of the tab & Displaying name of text editor & not the file being edited ) .

After some trial & error , i have been successful in getting the intended value through this command - ```xprop -id $(xdotool getwindowfocus) WM_CLASS | perl -pe 's/.*=\s"(.*?)".*/$1/g'```

Now the problem is how do i assign the value of this command to the ```label``` option in xwindow module config ?
If there's a better way of doing this , i am open for that as well

**Comment by patrick96** [comment]
That's exactly what the [script module](https://github.com/polybar/polybar/wiki/Module:-script) is for

**Reply by xd003** [reply]
Thanks for the heads up , i have managed to get it working perfectly but sadly its causing extremely high cpu usage ( almost 80% usage on idle without running any app ) .

Here's the script file named poly-title
```
#!/bin/bash

stitle="$(xprop -id $(xdotool getwindowfocus) | grep WM_CLASS | perl -pe 's/.*=\s"(.*?)".*/$1/g')"
if [ -n "$stitle" ]
then
  echo "$stitle"
else
  echo ""
fi
```

& here's the module 
```
[module/poly-title]
type = custom/script
exec = /home/xd003/.local/bin/poly-title
tail = true
```
Any clue whats causing this issue ? Would appreciate inputs/solution , using polybar for very first time

**Reply by patrick96** [reply]
With `tail = true`, the interval is 0 seconds. So the script is constantly being run. Remove the tail setting, it's only for scripts continuously producing output. And set the `interval` to an appropriate value

**Reply by xd003** [reply]
Oh but since it's supposed to ideally change the title every time the app focus is changed , i think updating it at a given interval won't be ideal , can't it be triggered on focus change only ?
Also how does xwindow work , haven't seen interval being defined there & my CPU usage was pretty much fine using it

**Reply by patrick96** [reply]
The script module only does interval updates, everything else you have to do yourself.

The xwindow module connects to the X server and listens for events. You can probably do a similar thing with `xprop -spy` but it may get a little more involved
